### PR TITLE
Adding MobileSAM

### PR DESCRIPTION
**Pull request by claydugo**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by claydugo** [comment]
@hmaarrfk

File: recipes/MobileSAM/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/ChaoningZhang/MobileSAM.git
+  git_rev: 12d80d4e32b277de299130d8ce28cc949fb54b6c
+  folder_name: MobileSAM
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:11]:
```suggestion
```

File: recipes/MobileSAM/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{% set name = "mobilesam" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/ChaoningZhang/MobileSAM.git
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:9]: Maybe you can post an issue upstream?

**Comment by hmaarrfk** [comment]
Hey @claydugo if there isn't an official release i'm tempted to say that we should leave this for our own private channel

**Comment by hmaarrfk** [comment]
I would discuss with the author to get an official license, and a release (maybe pypi) or just a tag.

File: recipes/MobileSAM/meta.yaml (hunk 2 above)
Comment (claydugo) [reply recipes/MobileSAM/meta.yaml:9]: [Posted](https://github.com/ChaoningZhang/MobileSAM/issues/122), hopeful for a response

File: recipes/MobileSAM/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  git_rev: 12d80d4e32b277de299130d8ce28cc949fb54b6c
+
+build:
+  number: 0
+  script: python -m pip install . -vv
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:14]:
```suggestion
  noarch: python
  script: python -m pip install . -vv
```

File: recipes/MobileSAM/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+requirements:
+  host:
+    - python
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:18]:
```suggestion
    - python >=3.9
```

File: recipes/MobileSAM/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    - pip
+    - pytorch
+    - timm
+
+  run:
+    - python
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:24]:
```suggestion
    - python >=3.9
```

File: recipes/MobileSAM/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+{% set name = "mobilesam" %}
+{% set version = "0.1.0" %}
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:2]:
```suggestion
{% set version = "0.0.0.20231030.12d80d4e32b277de299130d8ce28cc949fb54b6c" %}
```

File: recipes/MobileSAM/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{% set name = "mobilesam" %}
+{% set version = "0.0.0.20231030.12d80d4e32b277de299130d8ce28cc949fb54b6c" %}
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:2]:
```suggestion
{% set version = "1.0.0.20231030.12d80d4e32b277de299130d8ce28cc949fb54b6c" %}
```
they seem to call their version "1.0"

File: recipes/MobileSAM/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/ChaoningZhang/MobileSAM.git
+  git_rev: 12d80d4e32b277de299130d8ce28cc949fb54b6c
+
+build:
+  number: 0
+  noarch: python
+  script: python -m pip install . -vv
```
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

the example suggests

File: recipes/MobileSAM/meta.yaml
Comment (hmaarrfk) [inline recipes/MobileSAM/meta.yaml:1]: I think that the folder should be lowercase

File: recipes/mobilesam/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python >=3.9
+    - pip
+    - pytorch
+    - timm
```
Comment (hmaarrfk) [inline recipes/mobilesam/meta.yaml:22]:
```suggestion
    - timm
```

Comment (hmaarrfk) [inline recipes/mobilesam/meta.yaml:22]:
```suggestion
```

Comment (hmaarrfk) [reply recipes/mobilesam/meta.yaml:22]: it s eems to work without it.

during setup, only the `setup.py` file is called, so you can inspect it to look at dependencies

File: recipes/mobilesam/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,43 @@
+{% set name = "mobilesam" %}
```
Comment (hmaarrfk) [inline recipes/mobilesam/meta.yaml:1]:
```suggestion
```

File: recipes/mobilesam/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,42 @@
+{% set version = "1.0.0.20231030.12d80d4e32b277de299130d8ce28cc949fb54b6c" %}
+
+package:
+  name: mobilesam
```
Comment (hmaarrfk) [inline recipes/mobilesam/meta.yaml:4]: just falgging this, but the import name of the package is `mobile_sam` the author will likely make a choice on what they name their pypi package, we should try to follow suite.

File: recipes/mobilesam/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,42 @@
+{% set version = "1.0.0.20231030.12d80d4e32b277de299130d8ce28cc949fb54b6c" %}
```
Comment (hmaarrfk) [inline recipes/mobilesam/meta.yaml:1]:
```suggestion
{% set git_rev = "12d80d4e32b277de299130d8ce28cc949fb54b6c" %}
{% set version = "1.0.0.20231030." + git_rev[:8] %}
```
we can limit the sha a little bit.

File: recipes/mobilesam/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+package:
+  name: mobilesam
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/ChaoningZhang/MobileSAM.git
+  git_rev: 12d80d4e32b277de299130d8ce28cc949fb54b6c
```
Comment (hmaarrfk) [inline recipes/mobilesam/meta.yaml:9]:
```suggestion
  git_rev: {{ git_rev }}
```

**Pull request closed without merging**